pkg/skaffold/event/server.go

```diff
@@ -62,14 +65,19 @@ func (s *server) Handle(ctx context.Context, event *proto.Event) (*empty.Empty,
 }
 
 // newStatusServer creates the grpc server for serving the state and event log.
-func newStatusServer(port string) (func() error, error) {
-	if port == "" {
+func newStatusServer(originalPort int) (func() error, error) {
+	if originalPort == -1 {
 		return func() error { return nil }, nil
 	}
-	l, err := net.Listen("tcp", port)
+	port := util.GetAvailablePort(originalPort)
+	if port != originalPort && originalPort != constants.DefaultRPCPort {
```

Comment:
wouldn't  port != constants.DefaultRPCPort suffice?
is this to guard against condition where util.GetAvailbalePort returns -1?
Maybe we shd define an constant PORT_NOT_FOUND and assign it to -1 just to make this readable.

Reply:
this logic is basically "if the user provided a different port than the skaffold default, but we couldn't use that port because it was already in use, warn them that we're using a different port than they might expect".

if the user didn't provide a port, they probably don't care that skaffold didn't use its own default. the port being used will still be in the logs.

Reply:
ah.. ok.